### PR TITLE
Convert all errors in `@solana/rpc-subscriptions-*` to coded exceptions

### PR DESCRIPTION
**Pull request by steveluscher**

### CODE REVIEW

**Comment by steveluscher** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2236?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#2236** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2236?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#2235** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2235?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @steveluscher and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/rpc-subscriptions-transport-websocket/src/websocket-connection.ts [hunk 1]
```diff
@@ -150,8 +158,9 @@ export async function createWebSocketConnection({
                                     if (e === EXPLICIT_ABORT_TOKEN) {
                                         return;
                                     } else {
-                                        // TODO: Coded error.
-                                        throw new Error('WebSocket connection closed', { cause: e });
+                                        throw new SolanaError(SOLANA_ERROR__RPC_WEBSOCKET_TRANSPORT_CONNECTION_CLOSED, {
+                                            cause: e,
```
Comment (steveluscher) [inline packages/rpc-subscriptions-transport-websocket/src/websocket-connection.ts:162]: This is the reason for #2235.

File: packages/errors/src/codes.ts [hunk 2]
```diff
@@ -59,6 +59,9 @@ export const SOLANA_ERROR__MALFORMED_NUMBER_STRING = 52 as const;
 export const SOLANA_ERROR__TIMESTAMP_OUT_OF_RANGE = 53 as const;
 export const SOLANA_ERROR__RPC_TRANSPORT_HEADER_FORBIDDEN = 54 as const;
 export const SOLANA_ERROR__RPC_TRANSPORT_HTTP_ERROR = 55 as const;
+export const SOLANA_ERROR__RPC_WEBSOCKET_TRANSPORT_CLOSED_BEFORE_MESSAGE_BUFFERED = 56 as const;
+export const SOLANA_ERROR__RPC_WEBSOCKET_TRANSPORT_CONNECTION_CLOSED = 57 as const;
+export const SOLANA_ERROR__RPC_WEBSOCKET_TRANSPORT_FAILED_TO_CONNECT = 58 as const;
```
Comment (lorisleiva) [inline packages/errors/src/codes.ts:64]: How would you feel about prefixing RPC Subscriptions errors with `RPC_SUBSCRIPTIONS` and following the same convention as `RPC_TRANSPORT_HTTP` — i.e. the concrete name of the transport come after the abstract name of the interface. Something like:

```ts
// Before.
SOLANA_ERROR__RPC_WEBSOCKET_TRANSPORT_CLOSED_BEFORE_MESSAGE_BUFFERED
SOLANA_ERROR__RPC_WEBSOCKET_TRANSPORT_CONNECTION_CLOSED
SOLANA_ERROR__RPC_WEBSOCKET_TRANSPORT_FAILED_TO_CONNECT
SOLANA_ERROR__SUBSCRIPTION_EXPECTED_SERVER_SUBSCRIPTION_ID
SOLANA_ERROR__SUBSCRIPTION_CANNOT_CREATE_SUBSCRIPTION_REQUEST

// After.
SOLANA_ERROR__RPC_SUBSCRIPTIONS_TRANSPORT_WEBSOCKET_CLOSED_BEFORE_MESSAGE_BUFFERED
SOLANA_ERROR__RPC_SUBSCRIPTIONS_TRANSPORT_WEBSOCKET_CONNECTION_CLOSED
SOLANA_ERROR__RPC_SUBSCRIPTIONS_TRANSPORT_WEBSOCKET_FAILED_TO_CONNECT
SOLANA_ERROR__RPC_SUBSCRIPTIONS_EXPECTED_SERVER_SUBSCRIPTION_ID
SOLANA_ERROR__RPC_SUBSCRIPTIONS_CANNOT_CREATE_SUBSCRIPTION_REQUEST
```

I also think the first three should probably live under the subscription-related error codes. Unless I'm missing something.

**Comment by steveluscher** [comment]
### Merge activity

* **Mar 3, 7:02 PM PST**: @steveluscher started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2236).

**Pull request merged**